soco/data_structures_entry.py

```diff
@@ -19,6 +19,7 @@
 _LOG.debug("%s imported", __name__)
 
 
+@lru_cache()
```

Comment:
Tested and working


You can check how effective it is with
https://docs.python.org/3/library/functools.html#functools.lru_cache

```
To help measure the effectiveness of the cache and tune the maxsize parameter, the wrapped function is instrumented with a cache_info() function that returns a named tuple showing hits, misses, maxsize and currsize.
```

Reply:
I'm gathering some statistics and will share shortly. I'm also testing with adding a decorator to `parse_event_xml()` but that cache rate is a bit lower. Perhaps still useful, though.

Reply:
After a few hours in a recent test:
* `from_didl_string`: CacheInfo(hits=347, misses=263, maxsize=128, currsize=128)
* `parse_event_xml`: CacheInfo(hits=220, misses=776, maxsize=128, currsize=128)

Even though the `parse_event_xml` hit rate is lower, I think it's still worth adding.

Reply:
That's still a decent hit rate and xml is painfully slow to parse